### PR TITLE
ConvPackedParams: remove legacy format

### LINKED ISSUE #43649: placeholder to remove legacy conv packed params
https://github.com/pytorch/pytorch/pull/43086/ will change the ConvPackedParams serialization format.

This issue tracks deleting the legacy code after a reasonable forward compatibility period.

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a @vkuzo

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43651 ConvPackedParams: remove legacy format**

Summary:

This is a forward compatibility follow-up to
https://github.com/pytorch/pytorch/pull/43086/. We switch the
conv serialization to output the v2 format instead of the v1 format.

The plan is to land this 1 - 2 weeks after the base PR.

Test Plan:

```
python test/test_quantization.py TestSerialization.test_conv2d_graph_v2
python test/test_quantization.py TestSerialization.test_conv2d_nobias_graph_v2
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23355480](https://our.internmc.facebook.com/intern/diff/D23355480)

### CODE REVIEW

**Review by z-a-f — APPROVED** [review]
So the legacy format will be removed completely?

**Comment by vkuzo** [comment]
> So the legacy format will be removed completely?

this removes the code for models to save the legacy format.  The code to read the legacy format would still be there for awhile.

**Comment by raghuramank100** [comment]
Why do we need to land this later and not at the same time as the previous PR?

**Review by raghuramank100 — APPROVED** [review]

**Comment by vkuzo** [comment]
> Why do we need to land this later and not at the same time as the previous PR?

pretty much to handle this case:
* person A gets a dogfooding app on a mobile device with PyTorch revision A (dogfooding apps shipped once a week)
* person B creates an ML model on a server with PyTorch revision B

if revision A does not know how to handle models created with revision B, the behavior is undefined.  So, we first create revision A which knows how to read the new format but still uses the old format.  We then wait N days (to allow most people to get on revision A), and then delete the old path.

File: aten/src/ATen/native/quantized/cpu/conv_serialization.h [hunk 1]
```diff
@@ -167,9 +152,6 @@ ConvParamsSerializationType serialize_conv(
   auto output_padding = params->output_padding().vec();
   params_vec.insert(params_vec.end(), output_padding.begin(),
                     output_padding.end());
-  for (int i = 0; i < kSpatialDim; i++) {
```
Comment (vkuzo) [inline aten/src/ATen/native/quantized/cpu/conv_serialization.h:170]: Seems like this was left in https://github.com/pytorch/pytorch/pull/39714 by accident, and tests did not catch it.  This PR adds test coverage for this path, so we need to fix it to land.

**Pull request closed without merging**